fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreBase.java

```diff
@@ -80,9 +89,22 @@ public SubspaceProvider getSubspaceProvider() {
         return subspaceProvider;
     }
 
+    @Nonnull
+    public CompletableFuture<Subspace> getSubspaceAsync() {
+        if (subspaceFuture == null) {
+            subspaceFuture = subspaceProvider.getSubspaceAsync(context).whenComplete((s, e) -> {
+                subspace = s;
```

Comment:
Should this still set `subspace` if `e != null` (i.e., the future completes exceptionally?)?